File: tests/php/testcases/NextMonthTest.php
<?
require_once 'base.php';

class NextMonthTestCase extends TestServiceTestCase {

    public function test_null() {
        $this->assertEquals($this->service->next_month(null), null);
    }

    public function test() {
        $this->assertEquals(
            $this->service->next_month('2011-08-20'),
            '2011-09-20');
    }

    public function test_dt_null() {
        $this->assertEquals($this->service->next_month_dt(null), null);
    }

    public function test_dt() {
        $this->assertEquals(
            $this->service->next_month_dt('2011-08-20T00:04:59Z'),
            '2011-09-20T00:04:59Z');
    }
}

File: tests/mono/testcases/NextMonthTest.cs
using System;
using NUnit.Framework;

namespace TestCases {

    [TestFixture()]
    public class NextMonthTest : TestServiceTest {

        DateTime origin = new DateTime(1970, 1, 1, 0, 0, 0, 0);

        public DateTime from_timestamp(long timestamp) {
            return origin.AddSeconds(timestamp);
        }

        public long to_timestamp(DateTime date) {
            return (long)((date - origin).TotalSeconds);
        }

        [Test()]
        public void test_null() {
            Assert.AreEqual(null, service.next_month(null));
        }

        [Test()]
        public void test() {
            // The client sends datetime instead of just date.
            // There are two problems with such a behaviour:
            // 1. Need to parse datetime instaed of date, not a problem
            //    actually.
            // 2. It sends datetime in UTC timezone and it results in a shifted
            //    date value. And this is the problem.
        }

        [Test()]
        public void test_dt_null() {
            Assert.AreEqual(null, service.next_month_dt(null));
        }

        [Test()]
        public void test_dt() {
            Assert.AreEqual(
                new DateTime(2011, 9, 20, 7, 41, 25),
                service.next_month_dt(
                    new DateTime(2011, 8, 20, 7, 41, 25)).Value);
        }
    }
}
